File: app/com/github/eduardofcbg/plugin/es/ESModule.java
package com.github.eduardofcbg.plugin.es;

import play.api.Configuration;
import play.api.Environment;
import play.api.inject.Binding;
import play.api.inject.Module;
import scala.collection.Seq;

/**
 * Created by eduardo on 09-06-2015.
 */
public class ESModule extends Module {

    @Override
    public Seq<Binding<?>> bindings(Environment environment, Configuration configuration) {
        System.out.println("loading module");
        Config.setConfig(configuration);
        return seq(bind(ESComponent.class).to(ElasticSearchPlugin.class).eagerly());
    }

}

File: app/com/github/eduardofcbg/plugin/es/ESModule.scala
package com.github.eduardofcbg.plugin.es

import play.api.{Configuration, Environment}
import play.api.inject.{Binding, Module}

/**
 * Created by eduardo on 22-06-2015.
 */
class ESModule extends Module {

  override def bindings(environment: Environment, configuration: Configuration): Seq[Binding[_]] = {
    Seq(
      bind[ES].to[ESConfig].eagerly()
    )
  }

}
